Velocity_Model/z.py

```diff
@@ -0,0 +1,182 @@
+    )
+    return [os.path.join(vm_dir, basin) for basin in basin_list]
+
+
+v203_basin_list = get_basin_full_path(
+    [
+        "Data/NI_BASINS/Wellington_Polygon_Wainuiomata_WGS84.txt",
+        "Data/Boundaries/NewCanterburyBasinBoundary_WGS84_1m.txt",
+        "Data/Boundaries/BPVBoundary.txt",
+        "Data/SI_BASINS/Cheviot_Polygon_WGS84.txt",
+        "Data/SI_BASINS/Hanmer_Polygon_WGS84.txt",
+        "Data/SI_BASINS/Kaikoura_Polygon_WGS84.txt",
+        "Data/SI_BASINS/Marlborough_Polygon_WGS84_v0p1.txt",
+        "Data/SI_BASINS/Nelson_Polygon_WGS84.txt",
+        "Data/SI_BASINS/NorthCanterbury_Polygon_WGS84.txt",
+        "Data/Boundaries/WaikatoHaurakiBasinEdge_WGS84.txt",
+    ]
+)
+v204_basin_list = v203_basin_list + get_basin_full_path(
+    [
+        "Data/USER20_BASINS/WanakaOutlineWGS84.txt",
+        "Data/USER20_BASINS/WakatipuBasinOutlineWGS84.txt",
+        "Data/USER20_BASINS/alexandra_outline.txt",
+        "Data/USER20_BASINS/ranfurly_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_A_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_B_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_C_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_D_outline.txt",
+        "Data/USER20_BASINS/NE_otago/NE_otago_E_outline.txt",
+        "Data/USER20_BASINS/bal_outline_WGS84.txt",
+        "Data/USER20_BASINS/dun_outline_WGS84.txt",
+        "Data/USER20_BASINS/mos_outline_WGS84.txt",
+        "Data/USER20_BASINS/Murchison_Basin_Outline_v1_WGS84.txt",
+        "Data/USER20_BASINS/hakataramea_outline_WGS84.txt",
+        "Data/USER20_BASINS/waitaki_outline_WGS84.txt",
+    ]
+)
+v205_basin_list = v204_basin_list
+v206_basin_list = v205_basin_list + get_basin_full_path(
+    [
+        "Data/USER20_BASINS/SpringsJ_basin_outline_v1_WGS84.txt",
+        "Data/USER20_BASINS/Karamea_basin_outline_v1_WGS84.txt",
+        "Data/USER20_BASINS/CollingwoodBasinOutline_1_WGS84_v1.txt",
+        "Data/USER20_BASINS/CollingwoodBasinOutline_2_WGS84_v1.txt",
+        "Data/USER20_BASINS/CollingwoodBasinOutline_3_WGS84_v1.txt",
+    ]
+)
+basin_outlines = {
+    "2.03": v203_basin_list,
+    "2.04": v204_basin_list,
+    "2.05": v205_basin_list,
+    "2.06": v206_basin_list,
+}
+
+
+def extract_z(
+    z_types,
+    stat_df,
+    nzvm_path=Path("NZVM"),
+    version=2.06,
+):
+    """
+    Create a dataframe based on values from the NZVM
+
+    :param z_types:
+    :param stat_df: Dataframe containing stations - station should be the index and lon and lat should be the columns
+    :param nzvm_path:
+    :param version: NZVM version to extract values for
+    :return: Dataframe containing all z_types, station names and sigma
+    """
+    if len(stat_df) > MAX_NUM_GEN_MULTI_PROFILES:
+        print(
+            "Increase the MAX_NUM_GEN_MULTI_PROFILES constant in the VM code"
+            " (and the constant used in this check) and rerun"
```

Comment:
State the current value, and also say to recompile the NZVM (I assume it's a hardcoded value rather than in a config file)